### ISSUE TITLE
Android Builds Fail on `./configure` for libUnwind

### ISSUE THREAD

**Issue by zachriggle** [google/honggfuzz#159]
Specifically, it looks like libUnwind is trying to validate that the compiler works by building a simple binary and executing it.  Since we're building on Ubuntu, this fails.

```
$ make android-all
...
checking build system type... x86_64-unknown-linux-gnu
checking host system type... arm-unknown-linux-androideabi
checking target system type... arm-unknown-linux-androideabi
...
checking for arm-linux-androideabi-gcc... /usr/local/google/home/riggle/Downloads/android-ndk-r15c-linux-x86_64//toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/bin/arm-linux-androideabi-gcc --sysroot=/usr/local/google/home/riggle/Downloads/android-ndk-r15c-linux-x86_64//platforms/android-24/arch-arm
checking whether the C compiler works... no
configure: error: in `/media/riggle/big/master/vendor/google/tools/security/fuzzers/honggfuzz/third_party/android/libunwind':
configure: error: C compiler cannot create executables
```

It looks like the `configure` line is incorrect, from `config.log`:

```
It was created by libunwind configure 1.1, which was
generated by GNU Autoconf 2.69.  Invocation command line was

  $ ./configure --host=arm-linux-androideabi --disable-coredump
```

Which comes from `compile-libUnwind.sh`

```
./configure --host=$TOOLCHAIN --disable-coredump
```

We can see by adding `set -ex` to the top of the build script, that the issued command is:

```
+ ./configure --host=arm-linux-androideabi --disable-coredump
```

Ultimately I got this far before I realized that the issue is that the GCC path listed above doesn't actually exist -- I'd made a mistake extracting it.

I had made a mistake extracting the NDK and set the NDK path incorrectly.  However, I've already typed all of this up, so I'll just quietly request adding a `test -f` somewhere in the `Makefile` that checks whether the Android compiler actually exists before invoking the `third_party` build scripts.

Thanks!

**Comment by robertswiecki** [comment]
hi, @anestisb - would you be able to comment on this bug? I guess I know how to compile honggfuzz for Android, but the build system is still somewhat a mystery to me :)

If not, no worries, I'll try to take a look.

**Comment by anestisb** [comment]
Yup, checking it now.

**Comment by robertswiecki** [comment]
I'll let @anestisb comment on the PR, thanks.

One nit: I think -x would be better check for $CC/$CXX

**Comment by zachriggle** [comment]
Fixed to use `-x`

**Comment by robertswiecki** [comment]
Thanks, I'll wait for @anestisb ok and we can merge that :)

**Comment by anestisb** [comment]
The PR looks good to me. With this early error message it's more clear that is path related.
 
It's probably worth doing the same change to the capstone build script.

**Comment by robertswiecki** [comment]
Is it possible/worthwile to make it on the level of main Makefile?

**Comment by anestisb** [comment]
I think inside the bash scripts fit better, because the main honggfuzz modules follow different build approach than dependencies (ndk-build wrapper vs custom toolchain selection). I don't see any good reason to put more checks inside the master makefile. Master makefile will simply fail with a command not found if NDK is not configured properly and dependencies are already built.

Plus the fact that the external bash scripts that build the dependencies are structured to be standalone (some people, including me, might invoke them manually sometimes). So even placing the checks inside the master makefile, they would be probably required at the scripts level too.

So long story short, I don't think putting any further checks in the master makefile are necessary. If you think otherwise, it is indeed possible by applying similar compiler detection logic for the cross toolchains.

**Comment by zachriggle** [comment]
❤️
